NumLib/ODESolver/NonlinearSolver.cpp

```diff
@@ -45,6 +45,7 @@ bool NonlinearSolver<NonlinearSolverTag::Picard>::solve(
 
     bool error_norms_met = false;
 
+    MathLib::LinAlg::finalizeAssembly(x);
```

Comment:
@chleh Moving  MathLib::LinAlg::finalizeAssembly(x) to applyKnownSolutions() also solves the problem. By the way, set values individually inside   applyKnownSolutions()  is insufficiency for MPI computation, I want to add a function of setVectorValues in  MathLib